Sources/Swift/Tools/UIRedactBuilder.swift

```diff
@@ -44,23 +44,58 @@ class UIRedactBuilder {
     private var ignoreClassesIdentifiers: Set<ObjectIdentifier>
     ///This is a list of UIView subclasses that need to be redacted from screenshot
     private var redactClassesIdentifiers: Set<ObjectIdentifier>
-    
-    init() {
         
-        var redactClasses = [ UILabel.self, UITextView.self, UITextField.self ] +
-        //this classes are used by SwiftUI to display images.
-        ["_TtCOCV7SwiftUI11DisplayList11ViewUpdater8Platform13CGDrawingView",
-         "_TtC7SwiftUIP33_A34643117F00277B93DEBAB70EC0697122_UIShapeHitTestingView",
-         "SwiftUI._UIGraphicsView", "SwiftUI.ImageLayer", "UIWebView"
-        ].compactMap { NSClassFromString($0) }
+    /**
+     Initializes a new instance of the redaction process with specified options.
+
+     This initializer configures which `UIView` subclasses should be redacted from screenshots and which should be ignored during the redaction process.
+
+     - parameter options: A `SentryRedactOptions` object that specifies the configuration for the redaction process.
+     
+     - If `options.redactAllText` is `true`, common text-related views such as `UILabel`, `UITextView`, and `UITextField` are redacted.
+     - If `options.redactAllImages` is `true`, common image-related views such as `UIImageView` and various internal `SwiftUI` image views are redacted.
+     - The `options.ignoreRedactViewTypes` allows specifying custom view types to be ignored during the redaction process.
+     - The `options.redactViewTypes` allows specifying additional custom view types to be redacted.
+
+     - note: On iOS, views such as `WKWebView` and `UIWebView` are automatically redacted, and controls like `UISlider` and `UISwitch` are ignored.
+     */
+    init(options: SentryRedactOptions) {
+        var redactClasses = [AnyClass]()
+        
+        if options.redactAllText {
+            redactClasses += [ UILabel.self, UITextView.self, UITextField.self ]
+        }
+        
+        if options.redactAllImages {
+            //this classes are used by SwiftUI to display images.
+            redactClasses += ["_TtCOCV7SwiftUI11DisplayList11ViewUpdater8Platform13CGDrawingView",
+             "_TtC7SwiftUIP33_A34643117F00277B93DEBAB70EC0697122_UIShapeHitTestingView",
+             "SwiftUI._UIGraphicsView", "SwiftUI.ImageLayer"
+            ].compactMap(NSClassFromString(_:))
```

Comment:
`m`: I know this was already here before, but how did we come up with these class names? Doesn't this change from time to time? Can we add an explanation or a link so we can update it in the future?

Reply:
They should not change from time to time for the same class name. So this will be updated only if Apple start using other classes